### PR TITLE
Add aws_lambda_functions_with_full_privileges query for CloudFormation

### LINKED ISSUE #1392: Add aws_lambda_functions_with_full_privileges query for CloudFormation
### Platform
AWS CloudFormation

### Provider
AWS
### Description
AWS Lambda Functions should not have roles with polices granting full administrative privileges.

### PR DESCRIPTION
**Pull request by mcarvalhox**
AWS Lambda Functions should not have roles with polices granting full administrative privileges. This query verifies if attributes Effect, Action and Resources do so by stating certain values simultaneously. Closes #1392

### CODE REVIEW

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/metadata.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "aws_lambda_functions_with_full_privileges",
+  "queryName": "AWS Lambda Functions With Full Privileges",
+  "severity": "HIGH",
+  "category": "Identity and Access Management",
+  "descriptionText": "AWS Lambda Functions should not have roles with polices granting full administrative privileges.",
+  "descriptionUrl": "https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/quickref-lambda.html"
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/metadata.json:8]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/query.rego [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	endswith(statement.Resource,"*")
+}
+
+checkAction(statement) {
+	statement.Action[_] == "*"
+}
+	
+checkAction(statement) {
+	endswith(statement.Action[_],"*")
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/query.rego:47]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/negative.yaml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            Resource: arn:aws:iam::account-ID-without-hyphens:user/Bob
+Outputs:
+  AllSecurityGroups:
+    Description: Security Groups that are associated with the EC2 instance
+    Value:
+      Fn::Join:
+      - ", "
+      - Fn::GetAtt:
+        - AllSecurityGroups
+        - Value
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/negative.yaml:108]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/positive.yaml [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            Resource: arn:aws:logs:*:*:*
+Outputs:
+  AllSecurityGroups:
+    Description: Security Groups that are associated with the EC2 instance
+    Value:
+      Fn::Join:
+      - ", "
+      - Fn::GetAtt:
+        - AllSecurityGroups
+        - Value
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/positive.yaml:108]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/positive.yaml [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      Policies:
+      - PolicyName: root
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+          - Effect: Allow
+            Action:
+            - "*"
+            Resource: arn:aws:logs:*:*:*
+Outputs:
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/positive.yaml:100]: Remove the entire `Outputs` section to reduce sample size.

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/positive.yaml [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId:
+        Fn::FindInMap:
+        - AWSRegionArch2AMI
+        - Ref: AWS::Region
+        - Fn::FindInMap:
+          - AWSInstanceType2Arch
+          - Ref: InstanceType
+          - Arch
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/positive.yaml:74]: To reduce the sample size lets remove this entire section and leave the ImageId hardcoded with any value such as:
`ImageId: ami-0ff8a91507f77f867`

We just want this template to be syntactically valid.

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/positive.yaml [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+Mappings:
+  AWSInstanceType2Arch:
+    t2.micro:
+      Arch: HVM64
+    m1.small:
+      Arch: HVM64
+      AWSRegionArch2AMI:
+    us-east-1:
+      HVM64: ami-0ff8a91507f77f867
+      HVMG2: ami-0a584ac55a7631c0c
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/positive.yaml:24]: Let's remove this entire `Mappings`  section to reduce sample size.

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/negative.yaml [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        Ref: ExistingSecurityGroups
+      AppendedItem:
+        Ref: SecurityGroup
+  AppendItemToListFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Handler: index.handler
+      Role: !GetAtt LambdaExecutionRole.Arn
+      Code:
+        ZipFile: !Sub |
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/negative.yaml:56]: `!Sub` is not needed here. Remove the intrinsic function `!Sub` and leave the `|`

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/positive.yaml [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+        Ref: ExistingSecurityGroups
+      AppendedItem:
+        Ref: SecurityGroup
+  AppendItemToListFunction:
+    Type: AWS::Lambda::Function
+    Properties:
+      Handler: index.handler
+      Role: !GetAtt LambdaExecutionRole.Arn
+      Code:
+        ZipFile: !Sub |
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/positive.yaml:56]: `!Sub` is not needed here. Remove the intrinsic function `!Sub` and leave the `|`

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/negative.yaml [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId:
+        Fn::FindInMap:
+        - AWSRegionArch2AMI
+        - Ref: AWS::Region
+        - Fn::FindInMap:
+          - AWSInstanceType2Arch
+          - Ref: InstanceType
+          - Arch
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/negative.yaml:74]: To reduce the sample size lets remove this entire section and leave the ImageId hardcoded with any value such as:
`ImageId: ami-0ff8a91507f77f867`

We just want this template to be syntactically valid.

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/negative.yaml (hunk 3 above)
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/negative.yaml:108]: Remove the entire `Outputs` section to reduce sample size.

File: assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/negative.yaml [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+Mappings:
+  AWSInstanceType2Arch:
+    t2.micro:
+      Arch: HVM64
+    m1.small:
+      Arch: HVM64
+      AWSRegionArch2AMI:
+    us-east-1:
+      HVM64: ami-0ff8a91507f77f867
+      HVMG2: ami-0a584ac55a7631c0c
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/aws_lambda_functions_with_full_privileges/test/negative.yaml:24]: Let's remove this entire `Mappings`  section to reduce sample size.

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
Almost good

**Review by joaoReigota1 — APPROVED** [review]

**Review by rogeriopeixotocx — APPROVED** [review]

**Pull request merged**